native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -173,98 +166,54 @@ public static Object options(Environment env, BObject client, BString path, Obje
                                              BTypedesc targetType, String methodName) {
-        Object[] paramFeed = new Object[6];
+        Object[] paramFeed = new Object[3];
         paramFeed[0] = path;
-        paramFeed[1] = true;
-        paramFeed[2] = message;
-        paramFeed[3] = true;
-        paramFeed[4] = targetType;
-        paramFeed[5] = true;
+        paramFeed[1] = message;
+        paramFeed[2] = targetType;
         return invokeClientMethod(env, client, methodName, paramFeed);
     }
 
     private static Object invokeClientMethod(Environment env, BObject client, BString path, Object message,
                                              Object mediaType, Object headers, BTypedesc targetType,
                                              String methodName) {
-        Object[] paramFeed = new Object[10];
+        Object[] paramFeed = new Object[5];
         paramFeed[0] = path;
-        paramFeed[1] = true;
-        paramFeed[2] = message;
-        paramFeed[3] = true;
-        paramFeed[4] = targetType;
-        paramFeed[5] = true;
-        paramFeed[6] = mediaType;
-        paramFeed[7] = true;
-        paramFeed[8] = headers;
-        paramFeed[9] = true;
+        paramFeed[1] = message;
+        paramFeed[2] = targetType;
+        paramFeed[3] = mediaType;
+        paramFeed[4] = headers;
         return invokeClientMethod(env, client, methodName, paramFeed);
     }
 
     private static Object invokeClientMethod(Environment env, BObject client, String methodName, Object[] paramFeed) {
-        Future balFuture = env.markAsync();
-        Map<String, Object> propertyMap = getPropertiesToPropagate(env);
-        env.getRuntime().invokeMethodAsync(client, methodName, null, null, new Callback() {
-            @Override
-            public void notifySuccess(Object result) {
-                balFuture.complete(result);
+        return env.yieldAndRun(() -> {
+            try {
+                return env.getRuntime().callMethod(client, methodName, new StrandMetadata(true, null), paramFeed);
+            } catch (BError bError) {
+                return HttpUtil.createHttpError("client method invocation failed: " + bError.getErrorMessage(),
+                        HttpErrorType.CLIENT_ERROR, bError);
             }
-
-            @Override
-            public void notifyFailure(BError bError) {
-                BError invocationError =
-                        HttpUtil.createHttpError("client method invocation failed: " + bError.getErrorMessage(),
-                                                 HttpErrorType.CLIENT_ERROR, bError);
-                balFuture.complete(invocationError);
-            }
-        }, propertyMap, PredefinedTypes.TYPE_NULL, paramFeed);
-        return null;
-    }
-
-    private static Map<String, Object> getPropertiesToPropagate(Environment env) {
-        String[] keys = {CURRENT_TRANSACTION_CONTEXT_PROPERTY, KEY_OBSERVER_CONTEXT, SRC_HANDLER, MAIN_STRAND,
```

Comment:
With the new runtime APIs, don't we support adding properties to the strand which can be obtained `Environment.getStrandLocal` API? We have some logic based on these properties

Reply:
Do we have any tests covering the case? The strand local was not read anywhere.

Reply:
@TharmiganK  you can dd new properties to strandmetadata

Reply:
Here with new runtime APIs this code run in same strand so we do not need to copy properties. runtime will only run concurrently if it came from non strand thread.

Reply:
There are few use cases here: https://github.com/search?q=repo%3Aballerina-platform%2Fmodule-ballerina-http%20%22getStrandLocal%22&type=code

I think we are using it for access logging but we do not have any tests for that

Reply:
Still those tests will pass since this code is coming from extern function call, same strand will use so as properties.